azurerm/internal/services/datamigration/data_source_data_migration_service.go

```diff
@@ -0,0 +1,88 @@
+package datamigration
+
+import (
+	"fmt"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmDataMigrationService() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmDataMigrationServiceRead,
+
```

Comment:
We'll need to add timeouts to this resource

Reply:
I add a 30-5-30-30 minutes timeout to this resource. In my personal test, the creation takes longest, ~20min. There is no SLA for timeout in service document though.